include/libdnf/system/state.hpp

```diff
@@ -30,6 +32,41 @@ along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
 
 namespace libdnf::system {
 
+class PackageState {
+public:
+    bool userinstalled{false};
```

Comment:
The default value is only there so that the member variable is never uninitialized. In this case the intention is to always initialize the value in construction, but I wasn't sure what the public interface for this should be, so I didn't bother with an explicit constructor yet and relied on direct member initialization (or whatever it's called, I'm not sure right now).

I can add it if you insist, but this will need to be revised from API standpoint anyway, so I thought I'd just leave it for now.